nbclient/execute.py

```diff
@@ -575,6 +622,81 @@ def run_cell(self, cell, cell_index=0, store_history=False):
         # Return cell.outputs still for backwards compatibility
         return exec_reply, cell.outputs
 
+    async def async_run_cell(self, cell, cell_index=0, store_history=False):
+        parent_msg_id = self.kc.execute(
+            cell.source, store_history=store_history, stop_on_error=not self.allow_errors
+        )
+        self.log.debug("Executing cell:\n%s", cell.source)
+        exec_timeout = self._get_timeout(cell)
+        deadline = None
+        if exec_timeout is not None:
+            deadline = monotonic() + exec_timeout
+
+        cell.outputs = []
+        self.clear_before_next_output = False
+
+        # This loop resolves nbconvert#659. By polling iopub_channel's and shell_channel's
+        # output we avoid dropping output and important signals (like idle) from
+        # iopub_channel. Prior to this change, iopub_channel wasn't polled until
+        # after exec_reply was obtained from shell_channel, leading to the
+        # aforementioned dropped data.
+
+        # These two variables are used to track what still needs polling:
+        # more_output=true => continue to poll the iopub_channel
+        more_output = True
+        # polling_exec_reply=true => continue to poll the shell_channel
+        polling_exec_reply = True
+
+        while more_output or polling_exec_reply:
+            await asyncio.sleep(0)
```

Comment:
Out of curiosity, what is the purpose of this await?

Reply:
It just yields, so that the event loop can run other tasks.